### ISSUE TITLE
Ansi tests: last does not accept lisp integers (which cn be bignums)

### ISSUE THREAD

**Issue by kpoeck** [clasp-developers/clasp#462]
Examples:
````
(handler-case (last (list 1 2 3 4) most-positive-fixnum) (error (e) e))
-> #<CELL-ERROR>
CL-USER> (handler-case (last (list 1 2 3 4) (1+ (expt 2 60))) (error (e) e))
-> (4)
````

**Comment by kpoeck** [comment]
Perhaps something like this:
````
(defun last (list &optional (n 1))
	   (etypecase n
	     (fixnum (last-fixnum list n))
	     (bignum (last-bignum list n))))
````

**Comment by kpoeck** [comment]
I don't think the type can always be determined at compile time (just for constant values of n)

**Comment by Bike** [comment]
I seriously doubt lists with a bignum number of elements are in the cards, so last-bignum can just be a sign check on whether to return the whole list (positive) or signal a type error (negative).

**Comment by kpoeck** [comment]
Fixed by pr #498, closing
